components/operator/internal/resources/databases/watch.go

```diff
@@ -0,0 +1,32 @@
+package databases
+
+import (
+	"reflect"
+
+	"github.com/formancehq/operator/api/formance.com/v1beta1"
+	"github.com/formancehq/operator/internal/core"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func Watch(service string, target any) func(ctx core.Context, object *v1beta1.Database) []reconcile.Request {
```

Comment:
Is there a particular reason to use reflection here rather than generics? Something like: 

```go
func Watch[T client.Object](service string, target T) func(core.Context, *v1beta1.Database) []reconcile.Request {
    // ...
}
```